### PR TITLE
rustc_span: Optimize more hygiene operations using `Span::map_ctxt`

### PR DESCRIPTION
**Pull request by petrochenkov**
I missed these in https://github.com/rust-lang/rust/pull/125017.

### CODE REVIEW

File: compiler/rustc_span/src/lib.rs [hunk 1]
```diff
@@ -1065,33 +1065,41 @@ impl Span {
 
     #[inline]
     pub fn remove_mark(&mut self) -> ExpnId {
-        let mut span = self.data();
-        let mark = span.ctxt.remove_mark();
-        *self = Span::new(span.lo, span.hi, span.ctxt, span.parent);
+        let mut mark = ExpnId::root();
```
Comment (fbstj) [inline compiler/rustc_span/src/lib.rs:1068]: I guess this one is intentionally different from the others (which start as `None`), may it be documented?

Comment (petrochenkov) [reply compiler/rustc_span/src/lib.rs:1068]: Not sure what to document, these functions don't have their own logic and just forward to the underlying `SyntaxContext` methods, which may have different signatures and have their own documentation.

Comment (fbstj) [reply compiler/rustc_span/src/lib.rs:1068]: ahhh indeed, apologies, I didn't look at the the function signature

**Comment by rylev** [comment]
Improvements outweigh the regressions which all seemed to be reversed sometime soon after this landed. 

@rustbot label: +perf-regression-triaged

**Pull request merged**